inference-engine/src/inference_engine/threading/ie_cpu_streams_executor.cpp

```diff
@@ -66,37 +66,74 @@ struct CPUStreamsExecutor::Impl {
                     _impl->_streamIdQueue.pop();
                 }
             }
-            _numaNodeId = _impl->_config._streams
-                ? _impl->_usedNumaNodes.at(
-                    (_streamId % _impl->_config._streams)/
-                    ((_impl->_config._streams + _impl->_usedNumaNodes.size() - 1)/_impl->_usedNumaNodes.size()))
-                : _impl->_usedNumaNodes.at(_streamId % _impl->_usedNumaNodes.size());
 #if IE_THREAD == IE_THREAD_TBB || IE_THREAD == IE_THREAD_TBB_AUTO
-            auto concurrency = (0 == _impl->_config._threadsPerStream) ? tbb::task_arena::automatic : _impl->_config._threadsPerStream;
-            if (ThreadBindingType::NUMA == _impl->_config._threadBindingType) {
-#if TBB_INTERFACE_VERSION >= 11100  // TBB has numa aware task_arena api
-                _taskArena.reset(new tbb::task_arena{tbb::task_arena::constraints{_numaNodeId, concurrency}});
-#else
-                _taskArena.reset(new tbb::task_arena{concurrency});
-#endif
-            } else if ((0 != _impl->_config._threadsPerStream) || (ThreadBindingType::CORES == _impl->_config._threadBindingType)) {
-                _taskArena.reset(new tbb::task_arena{concurrency});
-                if (ThreadBindingType::CORES == _impl->_config._threadBindingType) {
-                    CpuSet processMask;
-                    int    ncpus = 0;
-                    std::tie(processMask, ncpus) = GetProcessMask();
-                    if (nullptr != processMask) {
-                        _observer.reset(new Observer{*_taskArena,
-                                                     std::move(processMask),
-                                                     ncpus,
-                                                     _streamId,
-                                                     _impl->_config._threadsPerStream,
-                                                     _impl->_config._threadBindingStep,
-                                                     _impl->_config._threadBindingOffset});
-                        _observer->observe(true);
+            #if TBB_INTERFACE_VERSION >= 12010 // TBB with hybrid CPU aware task_arena api
+            auto core_types = oneapi::tbb::info::core_types();
+            if (core_types.size() > 1 /*Hybrid CPU*/
```

Comment:
Currenty there is no plans on such a HW, but given the general open on how to (potentially) combine these (and future) settings- we may want to end up with accepting explicit cores mask, so that we create any combo.
E.g. smth like (tbb::info::mask doesn't exist yet)

auto **mask_big** = oneapi::tbb::info::mask(oneapi::tbb::info::core_types().back())
auto **mask_node0** = oneapi::tbb::info::mask( tbb::info::numa_nodes()[0]);
_taskArena.reset(new tbb::task_arena{tbb::task_arena::constraints{**mask_node0 & mask_big** , 0 /* concurency is deduced from the mask*/}});

We may even want to expose those masks (and let our exec_network API accept them). But this sounds too cumbersome for a regular user. Also makes the things too "hardcoded", as originally we use the TBB to affinitize quite "loosely"- e.g. **all** cores in a NUMA node, or **all**  Big (or little) cores .
Thoughts?